### DISCUSSION TITLE
I'm confuse about DeliverLast policy.

### DISCUSSION THREAD

**Discussion by nvcnvn** [nats-io/nats.go#1435] [Q&A]
On the official docs:
https://docs.nats.io/nats-concepts/jetstream/consumers#deliverpolicy
```
DeliverLast - When first consuming messages, the consumer will start receiving messages with the last message added to the stream, or the last message in the stream that matches the consumer's filter subject if defined.
```
Seem a little bit different with golang client docs:
https://github.com/nats-io/nats.go/blob/9e5e70676ec02a86975d09193c6dd0a95450800c/jetstream/consumer_config.go#L114
```
	// DeliverLastPolicy will start the consumer with the last sequence
	// received.
	DeliverLastPolicy
```
I'm not sure if this `DeliverLastPolicy` fit my use case here:
- I have 1 gateway instance that basically just receive a request for updating data, do some light verify then publish the message
- 3 "consumer_process" instances, listening for the messages and do the actual processing
  - the "consumer_process" instances MUST NOT process the same update request
  - all the "consumer_process" may be offline for a few minutes, when they start they should process the new request after last processed request

I think a simple limit-base stream is necessary since we want to "replay" the message some time (for internal testing).
Is `DeliverLastPolicy` the correct one I'm looking for?

**Comment by piotrpio** [comment]
Hello @nvcnvn 

The way `DeliverLastPolicy` works is when consuming messages on a consumer **for the first time**, the first message you get will be the last message which was added to the stream (unless you're using filter subjects, in which case that would be the last message on the stream with subject matching the filter).

As far as I understand it, this would not satisfy your use case, as this does not prevent the server from sending the same message to all listening consumers. What you can do, is simply use the same consumer across your consumer processes - then if you're consuming messages on each process, server will round-robin messages between processes.
